DMApp.Keystone/UserExecutable.cs

```diff
@@ -0,0 +1,140 @@
+            if (String.IsNullOrWhiteSpace(toolMetaData.ToolCommand))
+            {
+                string cleanedProgramName = Regex.Replace(programName, "[^a-zA-Z0-9 ]", "");
+                string cleanedCommandName = Regex.Replace(cleanedProgramName, @"\s+", "-").ToLower();
+                toolMetaData.ToolCommand = $"dataminer-keystone-{cleanedCommandName}";
+            }
+
+            if (String.IsNullOrWhiteSpace(toolMetaData.ToolVersion))
+            {
+                var exeVersion = fs.File.GetFileProductVersion(programPath);
+                if (String.IsNullOrWhiteSpace(exeVersion))
+                {
+                    exeVersion = fs.File.GetFileVersion(programPath);
+                }
+
+                if (String.IsNullOrWhiteSpace(exeVersion))
+                {
+                    exeVersion = "1.0.0";
+                }
+
+                toolMetaData.ToolVersion = exeVersion;
+            }
+
+            if (String.IsNullOrWhiteSpace(toolMetaData.Company)) toolMetaData.Company = "Undefined";
+            if (String.IsNullOrWhiteSpace(toolMetaData.Authors)) toolMetaData.Authors = "Undefined";
+
+            try
+            {
+                Dictionary<string, string> placeholdersToReplace = new()
+                {
+                    { nameof(ToolMetaData.ToolName), toolMetaData.ToolName },
+                    { nameof(ToolMetaData.ToolCommand), toolMetaData.ToolCommand },
+                    { nameof(ToolMetaData.Authors), toolMetaData.Authors },
+                    { nameof(ToolMetaData.Company), toolMetaData.Company },
+                    { nameof(ToolMetaData.ToolVersion), toolMetaData.ToolVersion },
+                    { "ProgramName", programName },
+                    { "ProgramNameShimmy", userProgramFolderName }
+                };
+
+                foreach (var topLevelFile in fs.Directory.EnumerateFiles(shimmyPathForPackaging))
+                {
+                    var extension = fs.Path.GetExtension(topLevelFile);
+                    if (extension != ".md" && extension != ".cs" && extension != ".txt" && extension != ".csproj") continue;
+
+                    string fileContent = fs.File.ReadAllText(topLevelFile);
+                    fileContent = ReplaceAllVariables(fileContent, placeholdersToReplace);
+                    fs.File.WriteAllText(topLevelFile, fileContent);
+                }
+
+                var packResult = dotnet.Run($"pack \"{shimmyPathForPackaging}/ExeShim.csproj\" --output \"{toolMetaData.OutputDirectory}\"");
+
+                if (!String.IsNullOrWhiteSpace(packResult.errors) || (packResult.output != null && packResult.output.Contains("error")))
+                {
+                    throw new InvalidOperationException("Failed to create dotnet tool");
+                }
+            }
+            finally
+            {
+                fs.Directory.DeleteDirectory(locationForPackaging);
```

Comment:
![image](https://github.com/SkylineCommunications/Skyline.DataMiner.CICD.Packages/assets/97458010/be4c4b1a-fe30-4c21-80c1-133ba5ee355a)

Currently the dotnet tool requires admin rights to be able to run. I think the best case would be to change the FileSystem so that the AllowWrites method needs to be called separately and not be part of the methods themselves.
And then see if it's possible to detect when the tool is run on GitHub or not.

Reply:
I'm not sure if it's possible to detect where a tool is running from in a stable and robust way. Running the tool As Administrator isn't that far of a stretch I think. Alternatively I suppose we could change the filesystem to 'best effort' it and 'gobble up' the exception and instead of crashing if the running user isn't privileged  to adjust the securityInfo.

Reply:
I've made a PR for the FileSystem: https://github.com/SkylineCommunications/Skyline.DataMiner.CICD.FileSystem/pull/18

Reply:
Updated AppPackageCreator and works